File: app/src/main/java/com/dhsdevelopments/potato/clientapi/channel2/Domain.java
package com.dhsdevelopments.potato.clientapi.channel2;

import com.google.gson.annotations.SerializedName;

import java.util.List;

public class Domain
{
    @SerializedName( "id" )
    public String id;

    @SerializedName( "name" )
    public String name;

    @SerializedName( "domain-type" )
    public String type;

    @SerializedName( "channels" )
    public List<Channel> channels;

    @Override
    public String toString() {
        return "Domain[" +
                       "id='" + id + '\'' +
                       ", name='" + name + '\'' +
                       ", type='" + type + '\'' +
                       ", channels=" + channels +
                       ']';
    }
}

File: app/src/main/java/com/dhsdevelopments/potato/clientapi/channel2/Domain.kt
package com.dhsdevelopments.potato.clientapi.channel2

import com.google.gson.annotations.SerializedName

class Domain {
    @SerializedName("id")
    lateinit var id: String

    @SerializedName("name")
    lateinit var name: String

    @SerializedName("domain-type")
    lateinit var type: String

    @SerializedName("channels")
    lateinit var channels: List<Channel>

    override fun toString(): String {
        return "Domain[id='$id', name='$name', type='$type', channels=$channels]"
    }
}
